schema/deploy/shipping/views.sql

```diff
@@ -2131,4 +2147,170 @@ grant select
 comment on view shipping.scan_return_results_v1 is
   'View of barcodes and presence/absence results for SCAN return of results on the UW Lab Med site';
 
+
+/******************** VIEWS FOR POWER BI DASHBOARDS ********************/
+create or replace view shipping.scan_demographics_v1 as
+
+    select
+        encountered_week,
+        sex,
+        age_range_fine,
+        age_range_fine_lower,
+        age_range_fine_upper,
+        race,
+        hispanic_or_latino,
+        income,
+        puma
+    from shipping.scan_encounters_v1
+;
+
+comment on view shipping.scan_demographics_v1 is
+  'A view of basic demographic data from the SCAN project for Power BI dashboards.';
+
+revoke all
+    on shipping.scan_demographics_v1
+  from "power-bi-exporter";
+
+grant select
+    on shipping.scan_demographics_v1
+    to "power-bi-exporter";
+
+
+create or replace view shipping.scan_hcov19_positives_v1 as
+
+    with hcov19_presence_absence as (
+        -- Collapse potentially multiple hCoV-19 results
+        select distinct on (sample_id)
+            sample.identifier as sample,
+            presence_absence_id,
+            pa.present as hcov19_present,
+            pa.created::date as hcov19_result_release_date
+        from
+            warehouse.presence_absence as pa
+            join warehouse.target using (target_id)
+            join warehouse.organism using (organism_id)
+            join warehouse.sample using (sample_id)
+        where
+            organism.lineage <@ 'Human_coronavirus.2019'
+            and pa.details @> '{"assay_type": "Clia"}'
+            and not control
+            -- We shouldn't be receiving these results from Samplify, but they
+            -- sometimes sneak in. Be sure to block them from this view so as
+            -- to not return inaccurate results to participants.
+            and target.identifier not in ('COVID-19_Orf1b', 'COVID-19-S_gene')
+        /*
+          Keep only the most recent push. According to Lea, samples are only
+          retested if there is a failed result. A positive, negative, or
+          indeterminate result would not be retested.
+
+          https://seattle-flu-study.slack.com/archives/CV1E2BC8N/p1584570226450500?thread_ts=1584569401.449800&cid=CV1E2BC8N
+        */
+        order by
+            sample_id, presence_absence_id desc
```

Comment:
ordering by sample_id, presence_absence_id desc  ... this probably means that the IDs increase with each new record.
How about ordering by sample_id, pa.created::date or some other "result date" instead to make it super obvious that we're taking the newest record? That is, use a date instead of an ID to make it clearer that we're ordering by a date.

Reply:
Ordering by date is definitely clearer, but I don't think it's reliable in our case. Each transaction of our presence/absence ETL creates multiple presence/absence records that will all have the same `created` timestamp. If there are duplicates within the same transaction, then we would have to order by `presence_absence_id` to get the newest record.

Reply:
I see. The created timestamp is not the time when the sample was resulted; it's when we imported it. Am I understanding what 'created' is?

Reply:
Yes, `created` is the timestamp of when the record was first processed by ID3C.

Reply:
Ahhh.